restli-tools/src/main/java/com/linkedin/restli/tools/data/SchemaFormatTranslator.java

```diff
@@ -345,46 +346,6 @@ private void writeTranslatedSchemasToDirectory(
     }
   }
 
-  private static NamedDataSchema checkForErrorsAndGetTopLevelSchema(
-      File file, String schemaFullname, PegasusSchemaParser parser, StringBuilder errorMessages)
-  {
-    StringBuilder errorMessageBuilder = parser.errorMessageBuilder();
-    if (errorMessageBuilder.length() > 0)
-    {
-      errorMessages.append("Failed to parse schema: ")
-          .append(file.getAbsolutePath())
-          .append("\nfullname: ")
-          .append(schemaFullname)
-          .append("\nerrors: ")
-          .append(errorMessageBuilder.toString())
-          .append("\n");
-      return null;
-    }
-    List<DataSchema> topLevelSchemas = parser.topLevelDataSchemas();
-    if (topLevelSchemas.size() != 1)
-    {
-      errorMessages.append("Expected one top level schema for: ")
-          .append(file.getAbsolutePath())
-          .append(" but got ")
-          .append(topLevelSchemas.size())
-          .append(" schemas.")
-          .append("\n");
-      return  null;
-    }
-    DataSchema sourceSchema = topLevelSchemas.get(0);
-    if (!(sourceSchema instanceof NamedDataSchema) ||
-        !((NamedDataSchema) sourceSchema).getFullName().equals(schemaFullname))
-    {
-      errorMessages.append("Parsed top-level schema does not match the schema file name. File: ")
```

Comment:
Is this error covered after this change?

Reply:
Yes, done here: https://github.com/linkedin/rest.li/blob/master/generator/src/main/java/com/linkedin/pegasus/generator/FileFormatDataSchemaParser.java#L177